src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -115,4 +127,263 @@ private boolean isSeqColL1Present(SeqColLevelOneEntity levelOneEntity) {
+
+        // "arrays" attribute
+        setArraysAttribute(seqColAEntity, seqColBEntity, comparisonResult);
+
+        // "elements"
+        List<String> seqColALengths = seqColAEntity.getLengths();
+        List<String> seqColBLengths = seqColBEntity.getLengths();
+        List<String> seqColANames = seqColAEntity.getNames();
+        List<String> seqColBNames = seqColBEntity.getNames();
+        List<String> seqColASequences = seqColAEntity.getSequences();
+        List<String> seqColBSequences = seqColBEntity.getSequences();
+
+        // "elements" attribute | "total"
+        setTotalElementsAttribute(seqColAEntity, seqColBEntity, comparisonResult);
+
+        // "elements" attribute | "a-and-b"
+        setA_and_B_ElementsAttribute(seqColAEntity, seqColBEntity, comparisonResult);
+
+        // "elements" attribute | "a-and-b-same-order"
+        // TODO: MAKE THIS GENERIC
+        // LENGTHS
+        if (!(lessThanTwoOverlappingElements(seqColALengths, seqColBLengths) && unbalancedDuplicatesPresent(seqColALengths, seqColBLengths))) {
+            boolean lengthsSameOrder = seqColALengths.equals(seqColBLengths);
+            comparisonResult.putIntoElements("a-and-b-same-order", "lengths", lengthsSameOrder);
+        } else {
+            comparisonResult.putIntoElements("a-and-b-same-order", "lengths", null);
+        }
+        // NAMES
+        if (!(lessThanTwoOverlappingElements(seqColANames, seqColBNames) && unbalancedDuplicatesPresent(seqColANames, seqColBNames))) {
+            boolean namesSameOrder = seqColANames.equals(seqColBNames);
+            comparisonResult.putIntoElements("a-and-b-same-order", "names", namesSameOrder);
+        } else {
+            comparisonResult.putIntoElements("a-and-b-same-order", "names", null);
+        }
+        // SEQUENCES
+        if (!(lessThanTwoOverlappingElements(seqColASequences, seqColBSequences) && unbalancedDuplicatesPresent(seqColASequences, seqColBSequences))) {
+            boolean sequencesSameOrder = seqColASequences.equals(seqColBSequences);
+            comparisonResult.putIntoElements("a-and-b-same-order", "sequences", sequencesSameOrder);
+        } else {
+            comparisonResult.putIntoElements("a-and-b-same-order", "sequences", null);
+        }
+
+        return comparisonResult;
+    }
+
+    private <T extends SeqColEntity> void setArraysAttribute(
+            T seqColAEntity, T seqColBEntity, SeqColComparisonResultEntity comparisonResult) {
+        Field[] seqColAFields = getFields(seqColAEntity);
+        Field[] seqColBFields = getFields(seqColBEntity);
+        List<Field> seqColCommonFields = getCommonFields(seqColAEntity, seqColBEntity);
+        List<String> seqColAFieldsNames = getFieldsNames(seqColAFields);
+        List<String> seqColBFieldsNames = getFieldsNames(seqColBFields);
+        List<String> commonFieldsNames = getFieldsNames(seqColCommonFields);
+        // a-only
+        comparisonResult.putIntoArrays("a-only", seqColAFieldsNames);
```

Comment:
I am a bit confused... shouldn't this `A minus (A intersect B)`?

Reply:
True, this is not correct.
I'll fix it.